smart-contract/program/src/processor/stake.rs

```diff
@@ -159,14 +159,15 @@ pub fn process_stake(
     }
 
     if stake_account.stake_amount > 0
-        && stake_account.last_claimed_time < stake_pool.header.last_crank_time
+        && stake_account.last_claimed_offset < stake_pool.header.current_day_idx as i64
     {
         return Err(AccessError::UnclaimedRewards.into());
     }
 
-    let current_time = Clock::get().unwrap().unix_timestamp;
+    let current_time = Clock::get()?.unix_timestamp;
+    let current_offset = (current_time - central_state.creation_time) / SECONDS_IN_DAY as i64;
```

Comment:
What do you think of moving these two lines into something like this in order to avoid repetition?

```rust
impl CentralState {
    pub fn get_current_offset(&self) -> u64 {
        let current_time = Clock::get().unwrap().unix_timestamp as u64;
        (current_time - self.creation_time as u64) / SECONDS_IN_DAY
    }
}
```

Reply:
Yes, why not :)